### PR TITLE
Demystify user storage

### PR DESCRIPTION
**Pull request by MKodde**
UserSecret storage was detached from the User storage. See commit details for details on what was causing unexpected behavior. The TLDR is that:

1. The shared construction logic for creating user storage left you with unexpected behavior
2. The user secret storage stored the secret storage internally. Making it a class with multiple responsibilities. :-1: 

An effort was put into covering the construction of the classes in test. Every type of user secret storage is now tested individually. The classes also no longer depend on the 'options' configuration parameter. They now have concrete dependency injection proof parameters. Making testing a breeze compared to the previous mess.

See: `https://www.pivotaltracker.com/story/show/181525762`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,4 +10,4 @@ install:
   - travis_retry composer install --no-interaction
 
 script:
-  - vendor/bin/phpunit
+  - vendor/bin/phpunit -c ./phpunit.xml --coverage-text
```
Comment (pmeulen) [inline .travis.yml]: You added this to a previous PR as well. Same comment here. Do we want this? Scrutenizer is currently setup to calculate code coverage.

**Review by pmeulen — APPROVED** [review]
Looks good. Nice spring cleaning :)

Let's discuss how to use travis/scutenizer (and what/when to move to GH actions). Otherwise fine!

File: .travis.yml (hunk 1 above)
Comment (MKodde) [reply .travis.yml]: That wat trickled down from the previous PR. I feel adding code coverage adds some runtime, and if Scrutenizer is going to cover this too. I'm ok with removing it here.

Comment (pmeulen) [reply .travis.yml]: OK, Let's remove coverage from travis then.

**Pull request merged**